### PR TITLE
feat: add basic autocomplete for powershell [WIP]

### PR DESCRIPTION
**Pull request by RodEsp**
## What does this PR do?
Adds very basic autocomplete functionality for powershell.
This will work when hitting `<tab>` right after typing the CLI name but won't work correctly if you hit `<tab>` after having typed in a topic or command.

e.g.
`myCli <tab>` will show you all the commands and nested commands for `myCli`
but `myCli topic <tab>` will not work appropriately.

@W-11791541@
@W-11791492@
Initial efforts for #2 

## Testing instructions
1. Clone this repo and pull down this branch
1. At the root of this repo on your local
   1. Run `yarn install && yarn build`
   1. Using an oclif CLI with [`@oclif/plugin-plugins`](https://github.com/oclif/plugin-plugins/) installed run `<cli> plugins link .`
1. Run `<cli> autocomplete powershell` and follow the instructions that are printed out
1. Type `<cli> <tab>` in your terminal and see autocomplete show up
1. Ensure it works as expected for powershell

### CODE REVIEW

**Review by mshanemc — APPROVED** [review]
approved these changes, but I don't like/understand the pre-existing code very much.

I agree with the decision not to mess with it, though.

File: src/commands/autocomplete/create.ts [hunk 1]
```diff
@@ -28,80 +27,43 @@ function sanitizeDescription(description?: string): string {
 export default class Create extends AutocompleteBase {
   static hidden = true
 
+  static args = [{name: 'shell', description: 'shell type', required: false}]
```
[the remark this answers is not in the document]
Comment (RodEsp) [reply src/commands/autocomplete/create.ts]: I updated the description to be more meaningful, lemme know if that's still not clear enough.

**Comment by RodEsp** [comment]
Heya @mshanemc this should be ready for another QA pass. The `&&` issue should be fixed 👍🏼 

In terms of this
> `NOTE: if your terminal starts as a login shell...` <- I have no idea if it does or not. I wonder if we could give better advice for novices

It's not a super straightforward thing to explain, this messaging has always been a part of the autocomplete command, not added in this PR. But if you want we could create a separate WI to address it.

**Comment by mshanemc** [comment]
QA

### zsh on macos (sfdx)
I ran `sfdx autocomplete`
then copy-pasted the output into my terminal:
`$ printf "eval $(sfdx autocomplete:script zsh)" >> ~/.zshrc; source ~/.zshrc`
then I ran `sfdx autocomplete zsh`
❓ I'm doing this step because that's how the instructions for Powershell read.  I'm not sure how a user would know whether `sfdx autocomplete`'s instructions are a prerequisite or or substitute for `sfdx autocomplete powershell` 

file now looks like this...is that what it should look like?)

```
eval 
SFDX_AC_ZSH_SETUP_PATH=/Users/shane.mclaughlin/Library/Caches/sfdx/autocomplete/zsh_setup && test -f $SFDX_AC_ZSH_SETUP_PATH && source $SFDX_AC_ZSH_SETUP_PATH; # sfdx autocomplete setup
eval 
SF_AC_ZSH_SETUP_PATH=/Users/shane.mclaughlin/Library/Caches/sf/autocomplete/zsh_setup && test -f $SF_AC_ZSH_SETUP_PATH && source $SF_AC_ZSH_SETUP_PATH; # sf autocomplete setupeval 

fpath=(
/Users/shane.mclaughlin/Library/Caches/sfdx/autocomplete/functions/zsh
$fpath
);
autoload -Uz compinit;
compinit;
```

✅ sfdx autocomplete seems to work great on zsh (commands, flags)
---

### mac/zsh/sf

autocomplete, copy-paste instructions results in an updated .zshrc
❌ autocomplete on `sf` doesn't work with spaces
✅ colons work fine (checked flags and commands)

**Comment by mshanemc** [comment]
QA (windows, pwsh, sfdx)

👎🏻 it was a bit odd seeing `$` at the beginning of the instructions.    I totally copy/pasted that in 😄 .  My pwsh prompt is `stuff>`.  Might be a small tweak to make it more windows friendly.

✅ `sfdx [anything]` will result in `sfdx alias:list` and pressing tab cycles through them quasi-alphabetically
❓  the aliased `auth` commands are showing up (as `force:auth` whatever)

**Comment by RodEsp** [comment]
> ### zsh on macos (sfdx)
> I ran `sfdx autocomplete` then copy-pasted the output into my terminal: `$ printf "eval $(sfdx autocomplete:script zsh)" >> ~/.zshrc; source ~/.zshrc` then I ran `sfdx autocomplete zsh` ❓ I'm doing this step because that's how the instructions for Powershell read. I'm not sure how a user would know whether `sfdx autocomplete`'s instructions are a prerequisite or or substitute for `sfdx autocomplete powershell`

There's a bug in oclif that causes it to not detect the shell you're running on correctly. This is why for now you have to pass in the shell you want to the `<myCli> autocomplete` command. When that is fixed the command should correctly detect the shell and you won't need to pass it in. We'll have to document that in `sf`/`sfdx` docs or the release notes.

> file now looks like this...is that what it should look like?)
> 
> ```
> eval 
> SFDX_AC_ZSH_SETUP_PATH=/Users/shane.mclaughlin/Library/Caches/sfdx/autocomplete/zsh_setup && test -f $SFDX_AC_ZSH_SETUP_PATH && source $SFDX_AC_ZSH_SETUP_PATH; # sfdx autocomplete setup
> eval 
> SF_AC_ZSH_SETUP_PATH=/Users/shane.mclaughlin/Library/Caches/sf/autocomplete/zsh_setup && test -f $SF_AC_ZSH_SETUP_PATH && source $SF_AC_ZSH_SETUP_PATH; # sf autocomplete setupeval 
> 
> fpath=(
> /Users/shane.mclaughlin/Library/Caches/sfdx/autocomplete/functions/zsh
> $fpath
> );
> autoload -Uz compinit;
> compinit;
> ```

You don't need the lines under the `eval` statement or the `eval` statements, those are just old artifacts of you having setup autocomplete before these changes.

> ### mac/zsh/sf
> autocomplete, copy-paste instructions results in an updated .zshrc ❌ autocomplete on `sf` doesn't work with spaces ✅ colons work fine (checked flags and commands)

Autocomplete for zsh isn't yet supported for CLIs with space separated commands, like `sf`, but that's not related to adding autocomplete on powershell.
 
> ### mac, bash, sfdx
> ❌ running the command `sfdx autocomplete` from a bash terminal results in zsh instructions (checked this in both vscode's integrated terminal and the default mac one). Maybe it's reading what my default shell is instead of the live shell? Screen Shot 2022-10-07 at 9 30 34 AM

This is due to the same bug I mentioned above, but that's not related to adding autocomplete on powershell. What I did add in this PR is the ability for users to pass in an argument to the `<myCLI> autocomplete` command to specify which shell they want (e.g. `<myCLI> autocomplete [bash|zsh|powershell]`)


> QA (windows, pwsh, sfdx)
>
> 👎🏻 it was a bit odd seeing $ at the beginning of the instructions. I totally copy/pasted that in 😄 . My pwsh prompt is stuff>. Might be a small tweak to make it more windows friendly.

Done, changed `$` to `>` for powershell instructions

❓ the aliased auth commands are showing up (as force:auth whatever)
Hmm... unsure what you mean by this. You mean that alises are showing up in autocomplete?

**Pull request merged**